internal/app/repository/service.go

```diff
@@ -0,0 +1,91 @@
+package repository
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"log/slog"
+	"net/http"
+
+	"github.com/joshsoftware/code-curiosity-2025/internal/config"
+	"github.com/joshsoftware/code-curiosity-2025/internal/repository"
+)
+
+type service struct {
+	repositoryRepository repository.RepositoryRepository
+	appCfg               config.AppConfig
+}
+
+type Service interface {
+	GetRepoByRepoId(ctx context.Context, githubRepoId int) (Repository, error)
+	FetchRepositoryDetails(ctx context.Context, getUserRepoDetailsUrl string) (FetchRepositoryDetailsResponse, error)
+	CreateRepository(ctx context.Context, repoGithubId int, repo FetchRepositoryDetailsResponse) (Repository, error)
+}
+
+func NewService(repositoryRepository repository.RepositoryRepository, appCfg config.AppConfig) Service {
+	return &service{
+		repositoryRepository: repositoryRepository,
+		appCfg:               appCfg,
+	}
+}
+
+func (s *service) GetRepoByRepoId(ctx context.Context, repoGithubId int) (Repository, error) {
+	repoDetails, err := s.repositoryRepository.GetRepoByGithubId(ctx, nil, repoGithubId)
+	if err != nil {
+		slog.Error("failed to get repository by github id")
+		return Repository{}, err
+	}
+
+	return Repository(repoDetails), nil
+}
+
+func (s *service) FetchRepositoryDetails(ctx context.Context, getUserRepoDetailsUrl string) (FetchRepositoryDetailsResponse, error) {
+	req, err := http.NewRequest("GET", getUserRepoDetailsUrl, nil)
+	if err != nil {
+		slog.Error("error fetching user repositories details", "error", err)
+		return FetchRepositoryDetailsResponse{}, err
+	}
+
+	req.Header.Add("Authorization", s.appCfg.GithubPersonalAccessToken)
+
+	client := &http.Client{}
```

Comment:
suggesstion -> Pass this in the service struct so that we can reuse it rather than creating a new client everytime